src/main/java/frc/robot/subsystems/shooter/Shooter.java

```diff
@@ -0,0 +1,145 @@
+package frc.robot.subsystems.shooter;
+
+import com.ctre.phoenix.motorcontrol.*;
+import com.ctre.phoenix.motorcontrol.can.TalonFX;
+import edu.wpi.first.wpilibj.controller.LinearQuadraticRegulator;
+import edu.wpi.first.wpilibj.estimator.KalmanFilter;
+import edu.wpi.first.wpilibj.system.LinearSystem;
+import edu.wpi.first.wpilibj.system.LinearSystemLoop;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpiutil.math.Matrix;
+import edu.wpi.first.wpiutil.math.Nat;
+import edu.wpi.first.wpiutil.math.VecBuilder;
+import edu.wpi.first.wpiutil.math.Vector;
+import edu.wpi.first.wpiutil.math.numbers.N1;
+import frc.robot.Constants;
+import frc.robot.Ports;
+import frc.robot.subsystems.UnitModel;
+import frc.robot.utils.MovingAverage;
+
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+import static frc.robot.Constants.Shooter.*;
+
+/**
+ * The shooter class represents the physical shooter.
+ * The purpose of the flywheel is to provide a set of functions to handle a shooting situation in a game.
+ *
+ * @author Barel
+ * @version 1.0
+ * @using TalonFX
+ * @since 2021
+ */
+public class Shooter extends SubsystemBase {
+    private final TalonFX main = new TalonFX(Ports.Shooter.MAIN);
+    private final TalonFX aux = new TalonFX(Ports.Shooter.AUX);
+    private final UnitModel unitModel = new UnitModel(TICKS_PER_ROTATION);
+    private final MovingAverage movingAverage;
```

Comment:
You would need multiple MovingAverage object. because each angle of the hood would need one

Reply:
I would change it when I start working on the adjustable hood.